rest/alert.go

```diff
@@ -0,0 +1,207 @@
+	}
+	alertListResp, ok := (*v).(*alertListResponse)
+	if !ok {
+		return nil, fmt.Errorf(
+			"incorrect value for v, expected value of type *[]*alerting.Alert, got: %T", v,
+		)
+	}
+	alertListResp.Results = append(alertListResp.Results, nextAlerts.Results...)
+	return resp, nil
+}
+
+// Get returns the details of a specific alert.
+//
+// NS1 API docs: https://ns1.com/api/#alert-alertid-get
+func (s *AlertsService) Get(alertID string) (*alerting.Alert, *http.Response, error) {
+	path := fmt.Sprintf("%s/%s/%s", alertingRelativeBase, "alerts", alertID)
+	req, err := s.client.NewRequest("GET", path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var alert alerting.Alert
+	resp, err := s.client.Do(req, &alert)
+	if err != nil {
+		switch err.(type) {
+		case *Error:
+			if resourceMissingMatch(err.(*Error).Message) {
+				return nil, resp, ErrAlertMissing
+			}
+		}
+		return nil, resp, err
+	}
+
+	return &alert, resp, nil
+}
+
+// Create takes a *alerting.Alert and creates a new alert.
+//
+// NS1 API docs: https://ns1.com/api/#alert-post
+func (s *AlertsService) Create(alert *alerting.Alert) (*http.Response, error) {
+	path := fmt.Sprintf("%s/%s", alertingRelativeBase, "alerts")
+	req, err := s.client.NewRequest("POST", path, &alert)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the alerts fields with data from api(ensure consistent)
+	resp, err := s.client.Do(req, &alert)
+	if err != nil {
+		switch err.(type) {
+		case *Error:
+			if err.(*Error).Resp.StatusCode == http.StatusConflict {
+				return resp, ErrAlertExists
+			}
+		}
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+// Update updates the fields specified in the passed alert object.
+//
+// NS1 API docs: https://ns1.com/api/#alert-alertid-patch
+func (s *AlertsService) Update(alert *alerting.Alert) (*http.Response, error) {
+	alertID := ""
+	if alert != nil && alert.ID != nil {
+		alertID = *alert.ID
+	}
+	path := fmt.Sprintf("%s/%s/%s", alertingRelativeBase, "alerts", alertID)
+
+	req, err := s.client.NewRequest("PATCH", path, &alert)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the alerts fields with data from api(ensure consistent)
+	resp, err := s.client.Do(req, &alert)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+// Replace replaces the values in an alert with the values in the passed object.
+//
+// NS1 API docs: https://ns1.com/api/#alert-alertid-put
+func (s *AlertsService) Replace(alert *alerting.Alert) (*http.Response, error) {
+	alertID := ""
+	if alert != nil && alert.ID != nil {
+		alertID = *alert.ID
+	}
+	path := fmt.Sprintf("%s/%s/%s", alertingRelativeBase, "alerts", alertID)
+
+	req, err := s.client.NewRequest("PUT", path, &alert)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the alerts fields with data from api (ensure consistent)
+	resp, err := s.client.Do(req, &alert)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+// Delete immediately deletes an existing alert.
```

Comment:
The word "immediately" might make my Spidey-Sense tingle if I was a customer reading this. Why do we say "immediately"? Is there a _non_-immediate way to delete an alert? Do other commands not work immediately? :laughing:

Reply:
I copied that from the notifier list sdk code - I was trying to blend in. Removed.